tests/test_core_network.py

```diff
@@ -3017,3 +3017,14 @@ def test_getting_link_slope_dictionary(network3):
     link_slope = (z_2 - z_1) / length
     slope_dict = network3.get_link_slope_dictionary(elevation_dictionary)
     assert slope_dict['0']['slope'] == link_slope
+
+
+def test_splitting_link_at_point(network1):
+    network1.split_link_at_point('0', 51.52287873320000, 0.14625948709420000)
+    correct_link_1_attributes = {'from': '101982', 'to': '101987', 'id': '1', 'freespeed': 4.166666666666667,
```

Comment:
this is a good place to use mocker, that way you will get the ids for the new node and two new links that you expect, but which are not part of the test. So this would look like this:

```python
def test_splitting_link_at_point(network1, mocker):
    new_node_ID = 'new_node_ID'
    new_link_1_ID = 'new_link_1_ID'
    new_link_2_ID = 'new_link_2_ID'
    mocker.patch.object(Network, 'generate_index_for_node', return_value=new_node_ID)
    mocker.patch.object(Network, 'generate_index_for_edge', return_value=new_link_1_ID)
    mocker.patch.object(Network, 'generate_index_for_edge', return_value=new_link_2_ID)
```
and then you can use the `new_node_ID, new_link_1_ID, new_link_2_ID` downstream. **I highly recommend this approach** - because we might change the ID generation methods one day, we've always been on the look out for something more robust and less random for a long time